src/features/Classes/ClassesFilters/index.jsx

```diff
@@ -0,0 +1,133 @@
+
+import { Col, Form } from '@edx/paragon';
+import { Select, Button } from 'react-paragon-topaz';
+import { logError } from '@edx/frontend-platform/logging';
+
+import { initialPage } from 'features/constants';
+import { fetchCoursesData } from 'features/Courses/data/thunks';
+import { fetchClassesData } from 'features/Classes/data/thunks';
+import { fetchInstructorsData } from 'features/Instructors/data/thunks';
+import { updateFilters, updateCurrentPage } from 'features/Licenses/data/slice';
+
+const notAssignedOption = {
+  label: 'Not assigned',
+  value: 'null',
+};
+
+const ClassesFilters = ({ resetPagination }) => {
+  const dispatch = useDispatch();
+  const institution = useSelector((state) => state.main.selectedInstitution);
+  const courses = useSelector((state) => state.courses.table.data);
+  const instructors = useSelector((state) => state.instructors.table.data);
+
+  const [courseOptions, setCourseOptions] = useState([]);
+  const [instructorOptions, setInstructorOptions] = useState([notAssignedOption]);
+  const [courseSelected, setCourseSelected] = useState(null);
+  const [instructorSelected, setInstructorSelected] = useState(null);
+
+  const isButtonDisabled = courseSelected === null && instructorSelected === null;
+
+  const handleSelectFilters = async (e) => {
+    e.preventDefault();
+
+    if (isButtonDisabled) {
+      return;
+    }
+
+    const form = e.target;
+    const formData = new FormData(form);
+    const formJson = Object.fromEntries(formData.entries());
+    const userName = `username=${formJson?.username === 'null' ? '' : formJson.username}`;
+    const instructor = `${instructorSelected?.value === 'null' ? 'null' : ''}`;
+
+    try {
+      dispatch(updateFilters(formJson));
+      dispatch(updateCurrentPage(initialPage));
+      dispatch(fetchClassesData(institution.id, initialPage, formJson?.course_name, userName, instructor, false));
+    } catch (error) {
+      logError(error);
+    }
+  };
+
+  const handleCleanFilters = () => {
+    dispatch(fetchClassesData(institution.id, initialPage));
+    resetPagination();
+    setInstructorSelected(null);
+    setCourseSelected(null);
+    dispatch(updateFilters({}));
+  };
+
+  useEffect(() => {
+    const parseCoursesToOptions = courses.length > 0
+      ? courses.map(course => ({
+        ...course,
+        label: course.masterCourseName,
+        value: course.masterCourseName,
+      })) : [];
+
+    const parseInstructorsToOptions = instructors.length > 0
+      ? instructors.map(instructor => ({
+        ...instructor,
+        label: instructor.instructorName,
+        value: instructor.instructorUsername,
+      }))
+      : [];
+
+    setCourseOptions(parseCoursesToOptions);
+    setInstructorOptions([notAssignedOption, ...parseInstructorsToOptions]);
+  }, [instructors, courses]);
+
+  useEffect(() => {
+    setInstructorSelected(null);
+    setCourseSelected(null);
+    dispatch(fetchCoursesData(institution.id, initialPage));
```

Comment:
I'm getting an error in the first render with the service because institution id is undefined, please add the validation that should be called only when that is not empty